File: Java/LinkedList/Hard/61-rotate-list/rotate-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode rotateRight(ListNode head, int k) {
        if (head == null || head.next == null || k == 0) return head;
        int len = 1;
        ListNode tail = head;
        while(tail.next !=null){
            len++;
            tail = tail.next;
        }
        k = k%len;
        if(k == 0)return head;
        else{
            tail.next = head;
            int n= len-k;
            ListNode temp = head;
            for(int i=1;i<n;i++){
                temp = temp.next;
            }
            head= temp.next;
            temp.next=null;
        } 
        return head;
    }
}

File: Python/Daily/April/61-rotate-list/rotate-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def rotateRight(self, head: Optional[ListNode], k: int) -> Optional[ListNode]:
        if head is None or head.next is None:
            return head
        n = 1
        temp = head
        while temp.next is not None:
            temp = temp.next
            n+=1
        # print(n)
        tail = temp
        k = k%n
        # print(k)
        if k==0:
            return head
        
        dist= n-k

        # print(dist)

        temp = head
        prev = None
        tail.next = head
        for i in range(dist):
            prev = temp
            temp = temp.next
        prev.next =None

        return temp
        
        

        